heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java

```diff
@@ -90,9 +97,13 @@ public Response uploadFile(
 
     String uploadDir = config.getStringValue(FILE_SYSTEM_DIRECTORY);
 
-    String fileName = UUID.randomUUID() + "-" + fileDetail.getFileName();
+    if (StringUtil.isNotBlank(getDownloadHostName())) {
```

Comment:
can we do a little refactoring and have method call getHostnameOrIp() so we are not setting a static variable like this. Then lines 119/120 becomes:
String uri = String.format("http://%s:%s/api/v1/file/download/%s", getHostnameOrIp(), getPort(), fileName);